update changelog (#483)
hashicorp/consul-k8s: CHANGELOG.md

  - The endpoints controller replaces the cleanup controller while preserving existing functionality. [[GH-476](https://github.com/hashicorp/consul-k8s/pull/476)], [[GH-454](https://github.com/hashicorp/consul-k8s/pull/454)]

  - Merged metrics configuration support is now partially managed by the endpoints controller.  [[GH-469](https://github.com/hashicorp/consul-k8s/pull/469)]

* Connect: Leader election support for connect webhook and controller deployment. [[GH-479](https://github.com/hashicorp/consul-k8s/pull/479)]

* Connect: Connect webhook no longer generates its own certificates and relies on them being provided as files on the disk. [[GH-454](https://github.com/hashicorp/consul-k8s/pull/454)]] 

* Connect: Connect pods and their Envoy sidecars no longer have a preStop hook as service deregistration is managed by the endpoints controller.  [[GH-467](https://github.com/hashicorp/consul-k8s/pull/467)]

BUG FIXES:
* CRDs: make lastSyncedTime a pointer to prevent setting last synced time Reconcile errors. [[GH-466](https://github.com/hashicorp/consul-k8s/pull/466)]

BREAKING CHANGES:

> * Connect: Kubernetes Services are now required for all connect injected applications.

Reviewer: Hi @kschoche, 

I am trying to understand where this change originated and having tracked it here from `consul-helm` I still can't seem to find where the requirement/change originated and why. I was hoping you might be able to point me in the right direction? Ultimately I would like to try and understand what benefits this brings, why the change was needed and what this means from a underlying discovery/routing perspective? I was quite the fan of not needing to write `Service` yaml for all services :disappointed: I also have a slight concern in that this change may introduce a requirement that a Consul service can no longer be deployed across a variety of infrastructure. I.e. At present I could have the same service deployed across Kubernetes, VMs, and another container orchestrator and requests will be load balanced across the varying deployments. Hopefully I am wrong, but this change triggered a warning bell for me that this may no longer be possible now.

Author: Hi! @onematchfox Thanks for pointing this out and I do apologize if things were not clear from the changelogs. We'll be sure to update them again with additional content to help clarify the changes as best as we can for the next beta release (coming soon).

With the exception of creating a Kubernetes service for your services registered in Kubernetes there should not be any breaking changes to your existing setups including how things work between VMs and Kubernetes or other orchestrators.

At a high level what has changed is that in order to support transparent proxy we needed to refactor *where* service registration/deregistration are managed in consul-k8s. We also needed to sync a ClusterIP to Consul for transparenty proxy support and this comes from the kubernetes service. This also enables you to use KubeDNS for service discovery if you wish.

With the addition of the Kubernetes service we're able to use a controller to trigger on service endpoint events which allows us to be more reactive and performant in how we manage syncing services, health checks, etc.
Doing this has also allowed us to also fix quite a few pain points that users had experienced with the existing codebase, such as the lifecycle sidecar no longer needs to be included to re-register services continuously and we no longer need to rely on preStop hooks to do deregistration which had caused stale service registrations in failure scenarios.

We'll have more docs/content up as we get closer to GA, please stay tuned and thank you for the feedback!

IMPROVEMENTS:
* CRDs: update the CRD versions from v1beta1 to v1. [[GH-464](https://github.com/hashicorp/consul-k8s/pull/464)]
* Connect: the connect-inject init container has been split into two init containers. [[GH-441](https://github.com/hashicorp/consul-k8s/pull/441)]

> * Connect: A new internal command `consul-k8s connect-init` has been added, it replaces the existing init container logic for ACL login and envoy bootstrapping as well as introduces a polling wait for service registration, see `Endpoints Controller` for more information. [[GH-446](https://github.com/hashicorp/consul-k8s/pull/446)], [[GH-452](https://github.com/hashicorp/consul-k8s/pull/452)], [[GH-459](https://github.com/hashicorp/consul-k8s/pull/459)]

Reviewer 2: ```suggestion
* Connect: A new internal command `consul-k8s connect-init` has been added. It replaces the existing init container logic for ACL login and Envoy bootstrapping and introduces a polling wait for service registration, see `Endpoints Controller` for more information. [[GH-446](https://github.com/hashicorp/consul-k8s/pull/446)], [[GH-452](https://github.com/hashicorp/consul-k8s/pull/452)], [[GH-459](https://github.com/hashicorp/consul-k8s/pull/459)]
```

> * Connect: A new controller `Endpoints Controller` has been added which is repsonsible for managing service endpoints and service registration. When a kubernetes service referencing a connect-injected deployment is deployed the endpoints controller will be responsible for managing the lifecycle of the connect-injected deployment. [[GH-455](https://github.com/hashicorp/consul-k8s/pull/455)], [[GH-467](https://github.com/hashicorp/consul-k8s/pull/467)], [[GH-470](https://github.com/hashicorp/consul-k8s/pull/470)], [[GH-475](https://github.com/hashicorp/consul-k8s/pull/475)]

Reviewer 2: ```suggestion
* Connect: A new controller `Endpoints Controller` has been added which is repsonsible for managing service endpoints and service registration. When a Kubernetes service referencing a connect-injected deployment is deployed the endpoints controller will be responsible for managing the lifecycle of the connect-injected deployment. [[GH-455](https://github.com/hashicorp/consul-k8s/pull/455)], [[GH-467](https://github.com/hashicorp/consul-k8s/pull/467)], [[GH-470](https://github.com/hashicorp/consul-k8s/pull/470)], [[GH-475](https://github.com/hashicorp/consul-k8s/pull/475)]
```
Super nit: capitalizing k for Kubernetes

  - This includes:
      - service registration and deregistration, formerly managed by the `init-container`.
      - monitoring health checks, formerly managed by `healthchecks controller`.
      - re-registering services in the events of consul agent failures, formerly managed by `consul sidecar` and the `cleanup controller`.

  - The endpoints controller replaces the health checks controller while preserving existing functionality. [[GH-472](https://github.com/hashicorp/consul-k8s/pull/472)]

  - The endpoints controller replaces the cleanup controller while preserving existing functionality. [[GH-476](https://github.com/hashicorp/consul-k8s/pull/476)], [[GH-454](https://github.com/hashicorp/consul-k8s/pull/454)]

  - Merged metrics configuration support is now partially managed by the endpoints controller.  [[GH-469](https://github.com/hashicorp/consul-k8s/pull/469)]

> * Connect: HA support for connect. [[GH-479](https://github.com/hashicorp/consul-k8s/pull/479)]

Reviewer 2: ```suggestion
* Connect: Leader election support for connect webhook and controller deployment. [[GH-479](https://github.com/hashicorp/consul-k8s/pull/479)]
```

> * Connect: Connect webhook no longer generate it's own certificates and relies on them being provided as files on the disk. [[GH-454](https://github.com/hashicorp/consul-k8s/pull/454)]] 

Reviewer 2: ```suggestion
* Connect: Connect webhook no longer generates its own certificates and relies on them being provided as files on the disk. [[GH-454](https://github.com/hashicorp/consul-k8s/pull/454)]] 
```

> * Connect: Envoy sidecar and pods no longer have a preStop hook as service deregistration is managed by the endpoints controller.  [[GH-467](https://github.com/hashicorp/consul-k8s/pull/467)]

Reviewer 2: ```suggestion
* Connect: Connect pods and their Envoy sidecars no longer have a preStop hook as service deregistration is managed by the endpoints controller.  [[GH-467](https://github.com/hashicorp/consul-k8s/pull/467)]
```

> * Connect: connect-inject webhook has been refactored to use controller-runtime manager. [[GH-454](https://github.com/hashicorp/consul-k8s/pull/454)]

Reviewer 3: Not sure we need this because the changelog is for user-facing functionality changes.

> 
> * Connect: connect-inject webhook has been refactored to use controller-runtime manager. [[GH-454](https://github.com/hashicorp/consul-k8s/pull/454)]

Author: ```suggestion
```

BUG FIXES:
* CRDs: make lastSyncedTime a pointer to prevent setting last synced time Reconcile errors. [[GH-466](https://github.com/hashicorp/consul-k8s/pull/466)]

BREAKING CHANGES:
* Connect: Kubernetes Services are now required for all connect injected applications.

> The kubernetes service name will be used as the service name to register with Consul unless the annotation `consul.hashicorp.com/connect-service` is provided to the deployment/pod to override this. If using ACLs the ServiceAccountName must match the service name used with Consul.

Reviewer 2: ```suggestion
The Kubernetes service name will be used as the service name to register with Consul unless the annotation `consul.hashicorp.com/connect-service` is provided to the deployment/pod to override this. If using ACLs the ServiceAccountName must match the service name used with Consul.
```
Super nit
